partner-chains-cli/src/deregister/mod.rs

```diff
@@ -0,0 +1,82 @@
+#[cfg(test)]
+mod tests;
+
+use crate::config::config_fields::{
+	CARDANO_COLD_VERIFICATION_KEY_FILE, CARDANO_PAYMENT_SIGNING_KEY_FILE,
+};
+use crate::config::{get_cardano_network_from_file, CHAIN_CONFIG_FILE_PATH, PC_CONTRACTS_CLI_PATH};
+use crate::io::IOContext;
+use crate::pc_contracts_cli_resources::establish_pc_contracts_cli_configuration;
+use crate::{smart_contracts, CmdRun};
+use anyhow::anyhow;
+
+#[derive(Debug, clap::Parser)]
+pub struct DeregisterCmd;
+
+impl CmdRun for DeregisterCmd {
+	fn run<C: IOContext>(&self, context: &C) -> anyhow::Result<()> {
+		let (chain_config, cardano_network) = read_chain_config_file(context)?;
+		context.print(
+		  &format!("This wizard will remove the candidate identified by given from the committee candidates of a chain identified by parameters:\n{}.\nCommittee Candidate Validator Address is {}\n",
+          &chain_config.chain_parameters,
+          &chain_config.cardano_addresses.committee_candidates_address)
+    	);
+		let (payment_signing_key_path, cold_vkey) = get_cardano_keys(context)?;
+		let pc_contracts_cli_resources = establish_pc_contracts_cli_configuration(context)?;
+		let output = context.run_command(&format!(
+			"{PC_CONTRACTS_CLI_PATH} deregister --network {} --ada-based-staking --spo-public-key {} {} {}",
+			cardano_network.to_network_param(),
+			cold_vkey,
+			smart_contracts::sidechain_params_arguments(&chain_config.chain_parameters),
+			smart_contracts::runtime_config_arguments(
+				&pc_contracts_cli_resources,
+				&payment_signing_key_path
+			)
+		))?;
+		// Smart Contracts CLI return code is 0 even if the deregistration fails, so we need to check the output text.
+		if output.contains("transactionId") {
```

Comment:
I kind of like combinators over explicit `if else` but I don't have strong opinions:

```Rust
output
    .contains("transactionId")
    .then_some(context.print(&format!("Deregistration successful: {}", output)))
    .ok_or_else(|| anyhow!("Deregistration failed: {}", output))
```

Reply:
I would use combinator if Rust had a proper one.
In Scala I would write
```
Either.cond(
  output.contains("transactionId"),
  context.print(&format!("Deregistration successful: {}", output)),
  anyhow!("Deregistration failed: {}", output)
)
```